File: application/app/src/main/java/com/yahyeet/boardbook/model/mock/repository/MockGameTeamRepository.java
package com.yahyeet.boardbook.model.mock.repository;

import com.yahyeet.boardbook.model.entity.GameTeam;
import com.yahyeet.boardbook.model.repository.IGameTeamRepository;

import java.util.List;
import java.util.concurrent.CompletableFuture;
import java.util.stream.Collectors;

/**
 * Mock implementation of a game team repository
 */
public class MockGameTeamRepository extends AbstractMockRepository<GameTeam> implements IGameTeamRepository {
	@Override
	public CompletableFuture<List<GameTeam>> findTeamsByGameId(String id) {
		return CompletableFuture.supplyAsync(() ->
			mockDatabase
				.stream()
				.filter(team -> team.getGame().getId().equals(id))
				.collect(Collectors.toList())
		);
	}
}

File: application/app/src/main/java/com/yahyeet/boardbook/model/mock/repository/MockGameTeamRepository.kt
package com.yahyeet.boardbook.model.mock.repository

import com.yahyeet.boardbook.model.entity.GameTeam
import com.yahyeet.boardbook.model.repository.IGameTeamRepository
import java.util.concurrent.CompletableFuture
import java.util.stream.Collectors

class MockGameTeamRepository : AbstractMockRepository<GameTeam>(), IGameTeamRepository {
    override fun findTeamsByGameId(id: String): CompletableFuture<List<GameTeam>> {
        return CompletableFuture.supplyAsync { mockDatabase.stream().filter { team -> team.game!!.id == id }.collect<List<GameTeam>, Any>(Collectors.toList()) }
    }
}
